allennlp/common/registrable.py

```diff
@@ -52,13 +63,18 @@ def add_subclass_to_registry(subclass: Type[T]):
 
     @classmethod
     def by_name(cls: Type[T], name: str) -> Type[T]:
+        logger.info(f"instantiating registered subclass {name} of {cls}")
         if name not in Registrable._registry[cls]:
             raise ConfigurationError("%s is not a registered name for %s" % (name, cls.__name__))
         return Registrable._registry[cls].get(name)
 
     @classmethod
     def list_available(cls) -> List[str]:
         """List default first if it exists"""
+        # This is necessary because some of the subclasses may not have been loaded
+        # at this point, and they're not registered until they're loaded.
+        _load_module(cls)
```

Comment:
Why didn't we need this before, if we need it now?  (Not complaining, it's just not obvious to me, and I'm curious.)

Reply:
I added it because classes weren't being registered, but it seems like it works without it now, so I'll take it out